homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Sensor for displaying the number of result from Flume."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from pyflume import FlumeDeviceList, FlumeData
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Flume Sensor"
+
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+FLUME_TYPE_SENSOR = 2
+
+SCAN_INTERVAL = timedelta(minutes=1)
+MIN_TIME_BETWEEN_FORCED_UPDATES = timedelta(seconds=5)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=45)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Flume sensor."""
+    _username = config[CONF_USERNAME]
+    _password = config[CONF_PASSWORD]
+    _client_id = config[CONF_CLIENT_ID]
+    _client_secret = config[CONF_CLIENT_SECRET]
+    time_zone = str(hass.config.time_zone)
+    name = config[CONF_NAME]
+    flume_entity_list = []
+
+    flume_devices = FlumeDeviceList(_username, _password, _client_id, _client_secret)
+
+    try:
+        for device in flume_devices.device_list:
+            if device["type"] == FLUME_TYPE_SENSOR:
+                flume = FlumeData(
+                    _username,
+                    _password,
+                    _client_id,
+                    _client_secret,
+                    device["id"],
+                    time_zone,
+                    SCAN_INTERVAL,
+                )
+                flume_entity_list.append(FlumeSensor(flume, f"{name} {device['id']}"))
+    except KeyError:
+        _LOGGER.error("No Flume Devices Returned of Type: %s", FLUME_TYPE_SENSOR)
+    except AttributeError as attr_error:
```

Comment:
When does the attribute error happen?

Reply:
After review, I don't believe this will occur. I have removed it as well. It originally was added as I was troubleshooting some of the automated code check's and my exception was too broad causing pylint to fail.